src/watson/registration.py

```diff
@@ -26,6 +30,24 @@
 from watson.models import SearchEntry, has_int_pk
 
 
+class UniversalEncoder(DjangoJSONEncoder):
```

Comment:
I'd be happier using the django JSON encoder. Sure, it's not comprehensive, but if people want better JSON encoding then they can override the serialize_meta method. Our aim is to make JSON dumping entensible, not to necessarily extend it in any particular way.